app/controllers/providers_controller.rb

```diff
@@ -59,6 +59,14 @@ def publish
     redirect_to details_provider_recruitment_cycle_path(@provider.provider_code, @recruitment_cycle.year)
   end
 
+  def training_providers
+    if @current_user["admin"]
+      @providers = @provider.training_providers(recruitment_cycle_year: @recruitment_cycle.year)
```

Comment:
For anyone looking at this ... yes, this is a bit redundant, but it seems that this is the way the `custom_endpoints` work with `JsonApiClient`. Could be improved, but not here & now.